queue/queue.go

```diff
@@ -0,0 +1,47 @@
+package queue
+
+import (
+	beatsqueue "github.com/elastic/beats/v7/libbeat/publisher/queue"
```

Comment:
Why are you using this? Is this a temporary import and going to be removed in a later relase?
I was under the impression that we are trying to avoid importing from beats in our new repositories. If we want to reuse something from beats we should either move it to `elastic-agent-libs` or copy it to the new repo where it is used.

Reply:
The shipper is initially going to be built out of the beats outputs, queues, and the code that glues them together. That is a large chunk of code to move that at this point would just slow down development. I would like us to prioritize getting a working shipper prototype over the code reorganization right now.

I am in favour of moving all that code into `elastic-agent-libs` afterwards or perhaps in the background. That will make it clear that the code is used outside of just beats now. 

I will create an issue to migrate the code we end up using out of beats, which I assume will also be easier once we know what pieces we need to reuse.

Reply:
Yeah, there is still more pending work to clean up and modularize the queue and I expect that as that progresses it will probably migrate to `elastic-agent-libs`, it just wasn't part of the first draft.

Reply:
https://github.com/elastic/elastic-agent-shipper/issues/43